pkgs/applications/blockchains/go-ethereum.nix

```
@@ -1,9 +1,10 @@
-{ lib, stdenv, buildGoModule, fetchFromGitHub, libobjc, IOKit }:
+{ lib, stdenv, buildGoModule, fetchFromGitHub, libobjc ? null, IOKit ? null }:
```

```suggestion
{ lib, stdenv, buildGoModule, fetchFromGitHub, libobjc, IOKit }:
```
If this throws an error it is packaged incorrectly and does not work on Darwin. Did you inherit the packages correct?

---

@SuperSandro2000 

i couldn't reproduce the issue anymore, so i have dropped that commit from the PR. maybe it was fixed somewhere else since then?

FTR, i don't know what "inherit the packages correctly" means. this is what i use locally to invoke/test the build:

```
nix-build --pure -E "with import <nixpkgs> {}; callPackage /[...]nixpkgs/pkgs/applications/networking/bee/bee.nix {}"
```

---

You want to do something like https://github.com/NixOS/nixpkgs/blob/687c03a5d05cee0512ac1b9d8b9ce94e3d2f4f3a/pkgs/development/compilers/llvm/12/default.nix#L123-L126 in the callPackage of the package.

---

@SuperSandro2000 do i get it right: to use the git HEAD version of `go-ethereum` on my linux install, i need to mess with darwin stuff? that's rather unfortunate, because i have no clue about darwin, nor can i test it.

isn't there a way to get darwin completely out of the picture when we are not on darwin?